File: API/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entities
{
    [Table("Products")]
    public class Product
    {
        public int Id { get; set; }
        public bool IsUploaded { get; set; } = false;
        public string Title { get; set; }
        public string Brand { get; set; }
        public float BuyPrice { get; set; }
        public string ItemId { get; set; }
        public float SellPrice { get; set; }
        public string Seller { get; set; }
        public int QuantitySold { get; set; }
        public float Profit { get; set; } 
        public string Uploaded {get;set;}
        public ICollection<Photo> Photos { get; set; }
        public string Url { get; set; }
        public string ProductCategory { get; set; }
        public string DetailedDescription { get; set; }

        public AppUser AppUser { get; set; }
        public int AppUserId { get; set; }

    }
}

File: client/src/app/models/Product.ts
export interface Product {
    id: number;
    isUploaded: boolean;
    isChecked: boolean; // not in the API
    title: string;
    brand: string;
    buyPrice: number;
    sellPrice: number;
    itemId:string;
    photos?: any;
    seller: string;
    quantitySold: number;
    profit: number;
    uploaded: string;
    photoUrl:string;
    url: string;
    productCategory: string;
    detailedDescription: string;
}

export interface updatePhoto{
    productId: number;
    photoId: number;
}